analysis/trajectory.Rmd

```diff
@@ -10,8 +10,77 @@ bibliography: ../ref.bib
+Slingshot was designed to model developmental trajectories in scRNA-seq data and serve as a component in an analysis pipeline after dimensionality reduction and clustering. 
 
+We will use the 10X PBMC data in the Bioconductor data package `TENxPBMCData` that we have been using in previous chapters. 
+
+```{r, results='asis', echo=FALSE}
+extractCached("workflows/tenx-batch-pbmc/4k", "clustering", "pbmc4k")
+```
+
+```{r}
+pbmc4k
+```
+
+If you recall, we created our cluster labels and have three reduced dimensions in our `SingleCellExperiment` object. 
+
+```{r}
+head(pbmc4k$cluster)
+```
+
+```{r}
+reducedDimNames(pbmc4k)
+```
+
+At this point, we have everything we need to run slingshot. 
+This is a two-step process composed of identifying the global lineage structure with a cluster-based minimum spanning tree (MST) and fitting simultaneous principal curves to describe each lineage.
+
+These two steps can be run separately with the `getLineages()` and `getCurves()` functions, or together with the wrapper function, `slingshot()` (recommended). 
+Here, we use the wrapper function for the analysis of the single-trajectory dataset. 
+
+The `slingshot()` wrapper function performs both steps of lineage inference in a single call. 
+The necessary inputs are a reduced dimensional matrix of coordinates and a set of cluster labels. 
+These can be separate objects or elements contained in a `SingleCellExperiment` object.
+
+To run `slingshot()` with the dimensionality reduction produced by PCA and our cluster labels, we do the following:
+
+```{r}
+pbmc4k <- slingshot(pbmc4k, clusterLabels = 'cluster', reducedDim = 'PCA')
+```
+
+If no clustering results are provided, it is assumed that all cells are part of the same cluster and a single curve will be constructed. 
+If no dimensionality reduction is provided, slingshot will use the first element of the list returned by `reducedDims()`.
+
+The output is a `SingleCellExperiment` object with `slingshot` results incorporated. 
+Most of the output is added to the metadata in the form of a list and is accessible via `metadata(pbmc4k)$slingshot`.
+Additionally, all inferred pseudotime variables (one per lineage) are added to the `colData`. 
+To extract all `slingshot` results in a single object, we can use the `SlingshotDataSet()` function. 
+This can be useful for visualization, as several plotting methods for `SlingshotDataSet` objects are included with the package. Below, we visuzalize the inferred lineage for the single-trajectory data with points colored by pseudotime.
+
+```{r}
+summary(pbmc4k$slingPseudotime_1)
+```
+
+We can also plot the results: 
+
+```{r}
+colors <- colorRampPalette(brewer.pal(11,'Spectral')[-6])(100)
+plotcol <- colors[cut(pbmc4k$slingPseudotime_1, breaks=100)]
+
+plot(reducedDims(pbmc4k)$PCA, col = plotcol, pch=16, asp = 1)
```

Comment:
`reducedDim(pbmc4k, "PCA")` is more efficient. 

But it seems that one could more simply just use `plotPCA(pbmc4k)` and add ggplot lines to it.

Reply:
fixed